xbmc/addons/AddonDatabase.cpp

```diff
@@ -1284,3 +1284,32 @@ void CAddonDatabase::GetInstallData(const AddonInfoPtr& addon)
     CLog::Log(LOGERROR, "CAddonDatabase::{}: failed", __FUNCTION__);
   }
 }
+
+bool CAddonDatabase::AddInstalledAddon(const std::shared_ptr<CAddonInfo>& addon,
+                                       const std::string& origin)
+{
+  try
+  {
+    if (!m_pDB)
+      return false;
+    if (!m_pDS)
+      return false;
+
+    std::string now = CDateTime::GetCurrentDateTime().GetAsDBDateTime();
+    BeginTransaction();
```

Comment:
Why a transaction for a single simple sql statement?

Reply:
..same reason, because it is a copy. will remove the transaction

Reply:
okay.

Reply:
SyncInstalled does multiple INSERTs in a loop, thus a transaction makes sense there, btw.